### PR TITLE
Setup tests

### PR DESCRIPTION
**Pull request by IvanGrginovInf**
Same as https://github.com/infinum/eightshift-frontend-libs/pull/385 into correct branch

### CODE REVIEW

**Review by dingo-d — APPROVED** [review]
Looks good to me 👍🏼

**Review by goranalkovic-infinum — APPROVED** [review]

File: tests/unit/scripts/helpers/check-attr.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+it('tests checkAttr returns default empty value for a specific type when there is no manifest default', () => {
+	const accordionManifest = getComponentManifest('accordion');
+	const mockAttributes = {
+		componentName: 'accordion',
+	};
+
+	const accordionContent = checkAttr('accordionContent', mockAttributes, accordionManifest);
+	expect(accordionContent).toBe('');
+});
```
Comment (volfkarlo) [inline tests/unit/scripts/helpers/check-attr.test.js:45]: New line at the end

File: tests/data/block-attributes.js [hunk 2]
```diff
@@ -14,10 +14,13 @@ export const blockAttributes = {
...
+			'blockquoteBlockquoteIntroMockTypographyUse',
+			'blockquoteBlockquoteParagraphMockTypographyContent'
 		],
 		notExpected: [
-			'headingaaaContent',
+			'headingContent',
+			'cardHeadingContent',
 		]
 	},
 };
```
Comment (volfkarlo) [inline tests/data/block-attributes.js:26]: New line at the end

File: tests/data/props-output.js [hunk 3]
```diff
@@ -12,10 +16,12 @@ export const propsOutput = {
 				],
 				notExpected: [
 					'accordionAccordionContent',
-				]
+				],
+				prefix: "accordion"
 			}
 		}
 	},
+	// Mock blockquote block
```
Comment (volfkarlo) [inline tests/data/props-output.js:24]:
```suggestion
	// Mock blockquote block.
```

File: tests/data/props-output.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
 /**
- * Define what props should or should not return on each lvl of a dependency tree, using the newName to identify 
+ * Define what props should or should not return on each level of a dependency tree, using the newName to identify 
  * dependencies.
+ *
+ * Ideally all blocks from the project should have a top level entry here with
+ * their dependency tree and `expected` / `notExpected` attributes and `prefix` defined.
  */
 export const propsOutput = {
+	// Accordion block
```
Comment (volfkarlo) [inline tests/data/props-output.js:9]:
```suggestion
	// Accordion block.
```

File: tests/data/props-output.js [hunk 5]
```diff
@@ -85,8 +92,9 @@ export const propsOutput = {
 					'cardButtonContent',
 				],
 				notExpected: [
-					'blockquoteCardImageUrl',
+					'blockquoteCardImageUrl'
```
Comment (volfkarlo) [inline tests/data/props-output.js:95]:
```suggestion
					'blockquoteCardImageUrl',
```
Didn't we say that there should be trailing comma

File: tests/helpers/blocks.js [hunk 6]
```diff
@@ -75,10 +85,12 @@ export const getComponentDependencies = (componentManifests, componentName) => {
  * Returns an object composed of all component dependencies (if they exist) if the following format:
  * {
  * 	 newName: realName
+ *   ...
```
Comment (volfkarlo) [inline tests/helpers/blocks.js:88]: Strange indent.

File: tests/unit/blocks/block-registration.test.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+ */
+const debugName = (builtProps) => {
+	return `"${builtProps.realName}" (newName: "${builtProps.newName || builtProps.realName}") component`;
+}
+
+/**
+ * Asserts that built props match the expected output.
+ *
+ * @param {object} expectedProps propsOutput object for specific component
+ * @param {object} builtProps Props built for this object.
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:28]: I'm not sure what is correct. I believe with dot at the end but we should check that info.

I've noticed it few times in this PR
```suggestion
 * @param {object} expectedProps propsOutput object for specific component.
 * @param {object} builtProps Props built for this object.
```

File: tests/unit/blocks/block-registration.test.js [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+ */
+const recursiveAssertProps = (expectedProps, builtProps, isBlock = false) => {
+
+	// Some sanity check
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:32]: Intended comment or? Dot at the end.

```suggestion
	// Some sanity check
```

File: tests/unit/blocks/block-registration.test.js [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+		builtProps,
+		`Missing attributes key in built props for: ${debugName(builtProps)}`
+	).toHaveProperty('attributes');
+
+	expect(
+		typeof builtProps.attributes,
+		`Attributes key found but is not an array in built props for: ${debugName(builtProps)}`
+	).toBe('object');
+
+	// Check expected
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:43]:
```suggestion
	// Check expected.
```

File: tests/unit/blocks/block-registration.test.js [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+	if (!isBlock || expectedProps.expected) {
+		for (const expected of expectedProps.expected) {
+			expect(
+				builtProps.attributes,
+				`Missing expected property in built props for: ${debugName(builtProps)}`
+			).toHaveProperty(expected);
+		}
+	}
+
+	// Check not expected
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:53]:
```suggestion
	// Check not expected.
```

File: tests/unit/blocks/block-registration.test.js [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+	if (!isBlock || expectedProps.notExpected) {
+		for (const notExpected of expectedProps.notExpected) {
+			expect(
+				builtProps.attributes,
+				`Found not-expected property in built props for: ${debugName(builtProps)}`
+			).not.toHaveProperty(notExpected);
+		}
+	}
+
+	// Check prefix (but only if expected)
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:63]:
```suggestion
	// Check prefix (but only if expected).
```

File: tests/unit/blocks/block-registration.test.js [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+			`Missing "prefix" key in built props for ${debugName(builtProps)}`
+		).toHaveProperty('prefix');
+
+		expect(
+			builtProps.attributes.prefix,
+			`Prefix key not correct when sending props to ${debugName(builtProps)}`
+		).toEqual(expectedProps.prefix);
+	}
+
+	// Repeat if expected has subComponents
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:76]:
```suggestion
	// Repeat if expected has subComponents.
```

File: tests/unit/blocks/block-registration.test.js [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+	if (expectedProps.components && Object.keys(expectedProps.components).length > 0) {
+		for (const componentKey of Object.keys(expectedProps.components)) {
+
+			// Expect built props has subComponents
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:80]:
```suggestion
			// Expect built props has subComponents.
```

File: tests/unit/blocks/block-registration.test.js [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+				builtProps,
+				`Missing "subComponents" key in built props for ${debugName(builtProps)}. Expecting one because this has subComponents defined in props-output.js`
+			).toHaveProperty('subComponents');
+
+			expect(
+				builtProps.subComponents,
+				`"subComponents" key in built props empty for ${debugName(builtProps)}.`
+			).not.toEqual([]);
+
+			// Find the correct subComponent
```
Comment (volfkarlo) [inline tests/unit/blocks/block-registration.test.js:91]:
```suggestion
			// Find the correct subComponent.
```

**Review by volfkarlo — COMMENTED** [review]
Seems good!

Just some dots at the end of comments and similar stuff.

**Pull request merged**